x-pack/legacy/plugins/alerting/server/alerts_client.ts

```diff
@@ -215,6 +216,34 @@ export class AlertsClient {
     return this.getAlertFromRaw(id, updatedObject.attributes, updatedObject.references);
   }
 
+  public async updateApiKey({ id }: { id: string }) {
+    const {
+      references,
+      attributes: { enabled },
+    } = await this.savedObjectsClient.get('alert', id);
+
+    if (enabled === false) {
+      throw Boom.badRequest(`Can't update API key on a disabled alert`);
```

Comment:
Wondering what the rationale for this is.  Seems like I might want to take ownership of an alert and then enable it again.

Reply:
ya, that seems like a valid use case.  Especially if we ever get to the point of disabling "bad actor" alerts _somehow_, which doesn't seem out of the question.  Bob builds a bad alert, leaves the company, we auto-disable it, now no one can take ownership.

Reply:
There's current business logic to take ownership when enabling a disabled alert. Currently disabled alerts remove the `apiKey` and `apiKeyOwner` values until someone else re-claims it (by enabling the alert). My thought process was it wouldn't make sense to remove apiKey when disabling an alert but allow to generate a new api key here (that will then get overwritten later by the enable API call).

I'm happy to change the behaviour if it's not what you expect.

Reply:
Yeah I think we need to remove the limitation.  Can't think of anything good that comes of it, and it would reduce some of this code.

Reply:
I have fixed the code in this commit: https://github.com/elastic/kibana/pull/45925/commits/7fcb2b09da4e1b4b679916e87823a39c2bce030d.